python/lsst/ts/salobj/sal_info.py

```diff
@@ -644,38 +651,343 @@ async def _run_kafka(self) -> None:
         finally:
             await self.close(cancel_run_kafka_task=False)
 
+    def _blocking_setup_kafka(self) -> None:
+        """Set up Kafka.
+
+        Create topics in the Kafka broker.
+        Register topic schemas with the Kafka schema registry.
+        CCreate serializers and deserializers.
+        Create a consumer if there are any read topics.
+        Create a producer if there are any write topics.
+
+        Set the following attributes:
+
+        * _consumer, if there are any read topics
+        * _producer, if there are any write topics
+        * _deserializers_and_contexts
+        * _serializers_and_contexts
+        """
+        self._blocking_create_topics()
+        self._schema_registry_client = SchemaRegistryClient(
+            dict(url=self.schema_registry_url)
+        )
+        self._blocking_register_schema(
+            schema_registry_client=self._schema_registry_client
+        )
+        self._blocking_create_deserializers(
+            schema_registry_client=self._schema_registry_client
+        )
+        self._blocking_create_serializers(
+            schema_registry_client=self._schema_registry_client
+        )
+
+        self._blocking_create_producer()
+        self._blocking_create_consumer()
+
     def _blocking_create_topics(self) -> None:
-        """Create missing Kafka topics."""
-        # Dict of kafka topic name: topic_info
-        topic_info_dict = {
+        """Create missing Kafka topics for this SAL component."""
+        # A dict of kafka_name: topic_info.
+        # This elides duplicate names between self._read_topics
+        # and self._write_topics
+        topic_infos = {
             topic.topic_info.kafka_name: topic.topic_info
             for topic in itertools.chain(
                 self._read_topics.values(), self._write_topics.values()
             )
         }
-        new_topics_info = [
+        if not topic_infos:
```

Comment:
I see you are using this pattern here and in other places. Don't you think it is a bit counter-intuitive? 

I think doing

```
if something_is_true:
    make_something
```

is more readable than

```
if something_is_false:
    return

make_something
```

Reply:
I prefer this pattern because I like to get the error checking and sanity checking out of the way first, then write the complicated code for the normal case fully outdented.

I find it much easier to read than:
```
if everything is OK then:
   do a lot of stuff that takes up many lines of code
else:
  do a bit of error handling that only takes a few lines and is so far away from the initial "if"
  that the user has forgotten what conditions lead to this small block of code
```